gimie/project.py

```diff
@@ -68,79 +71,41 @@ def __init__(
         else:
             self.project_dir = path
 
-        self.extractors = self.get_extractors(sources)
+        self.extractor = get_extractor(
+            self.url,
+            git_provider,
+            base_url=self.base_url,
+            local_path=self.project_dir,
+        )
+        self.parsers = get_parsers(path, parser_names)
 
     def extract(self) -> Graph:
-        repos = [ex.extract() for ex in self.extractors]
-        graphs = [repo.to_graph() for repo in repos]
-        graph = combine_graphs(*graphs)
-        return graph
+        """Extract repository metadata from git provider to RDF graph and enrich with
+        metadata parsed from file contents."""
 
-    def get_extractors(self, sources: Iterable[str]) -> List[Extractor]:
+        repo = self.extractor.extract()
+        repo_graph = repo.to_graph()
 
-        extractors: List[Extractor] = []
-        for src in sources:
-            extractor = get_extractor(
-                self.url,
-                src,
-                base_url=self.base_url,
-                local_path=self.project_dir,
-            )
-            extractors.append(extractor)
+        files = self.extractor.list_files()
+        parsed_graphs = [
+            parser.parse_multiple(files) for parser in self.parsers
```

Comment:
would it make sense separating the identification of the file from the parsing of the file. 
I.e. for files: look for license file, if there is one, call the parser. 
the question is: is it the job of a parser to find a file ? 
what if there are thousand of files being handed over from parser to parser when all the parser cares about is the small license file?

Reply:
Excellent idea! We can use a [simple factory](https://stackoverflow.com/a/45394287/8440675) here.
It could be either a `get_parser(filename)` function or a `Parser.get_parser(filename)` static method.

Reply:
`gimie.parsers.select_parser(path) -> Parser` is now available to retrieve the correct parser based on input file path.